### PR TITLE
Add `...` to all vctrs generics

### LINKED ISSUE #399: Take `...` in all vctrs generics
This way their signature can be extended without causing downstream failures.

### PR DESCRIPTION
**Pull request by lionel-**
This makes it easier to extend the generics with new arguments without breaking backward compatibility. This is not to allow extensibility of methods, so we check the dots are empty before dispatch.

Closes #399

### CODE REVIEW

**Comment by lionel-** [comment]
Now also in the S3 vignette: https://github.com/r-lib/vctrs/commit/1b97ed2d

File: R/arith.R [hunk 1]
```diff
@@ -14,6 +14,8 @@
 #' @param op An arithmetic operator as a string
 #' @param x,y A pair of vectors. For `!`, unary `+` and unary `-`, `y` will be
 #'   a sentinel object of class `MISSING`, as created by `MISSING()`.
+#' @inheritParams ellipsis::dots_empty
```
Comment (hadley) [inline R/arith.R]: Does this do anything? The argument of `check_dots_empty()` is `env`?

Comment (lionel-) [reply R/arith.R]: This inherits from `dots_empty` which was written for this purpose: https://github.com/r-lib/ellipsis/blob/d8bf8a34eb3be04a52169bc69759ebccc7f247e2/R/dots.R#L6

Comment (hadley) [reply R/arith.R]: Ah got it.

**Pull request merged**